sdk/eventhub/event-hubs/src/eventHubProducerClient.ts

```diff
@@ -172,15 +184,36 @@ export class EventHubProducerClient {
    */
   async createBatch(options: CreateBatchOptions = {}): Promise<EventDataBatch> {
     throwErrorIfConnectionClosed(this._context);
+    const { enableIdempotentPartitions } = this._clientOptions;
+    const partitionId = isDefined(options.partitionId) ? String(options.partitionId) : undefined;
```

Comment:
So, I thought it might, but it actually can cause some issues since there are `partitionId != null` checks in downstream code that would branch differently with an empty string versus undefined (That may be a refactor for another day)

Reply:
Yeah, it's interesting the things you notice on re-review. :)